ArcGISRuntimeSDKQt_CppSamples/Routing/OfflineRouting/OfflineRouting.cpp

```diff
@@ -63,21 +63,28 @@ QString defaultDataPath()
 } // namespace
 
 OfflineRouting::OfflineRouting(QObject* parent /* = nullptr */):
-  QObject(parent)
+  QObject(parent),
+  m_stopsOverlay(new GraphicsOverlay(this)),
+  m_routeOverlay(new GraphicsOverlay(this))
 {
+  const QString folderLocation = QString("%1/ArcGIS/Runtime/Data/tpk/san_diego").arg(defaultDataPath());
+  if (!QFileInfo::exists(folderLocation))
+    qWarning() << "Please download required data.";
```

Comment:
do you need to bail out here?

Reply:
Do you think a qDebug is a better way to handle this @lsmallwood ?

Reply:
either `qWarning` or `qDebug` is ok i think - my comment was whether we should return immediately and not do the remaining c'tor logic...

Reply:
Oh yes I think it would be better to return immediately